docs/development.rst

```diff
@@ -152,26 +152,107 @@ The top-level document to open with a web browser will be
 Testing
 -------
 
-To run unit tests in the currently active Python environment, issue one of
-these example commands:
+All of the following `make` commands run the tests in the currently active
+Python environment. Depending on how the `pywbem` package is installed in that
+Python environment, either the directories in the main repository directory
+are used, or the installed package.
+
+The test case files and any utility functions they use are always used from
+the main repository directory, and are located in the `testsuite` directory
+structure:
 
 ::
 
-    $ make test                                              # Run all unit tests
-    $ PYTHONPATH=. py.test testsuite/test_cim_obj.py -s      # Run only this test source file
-    $ PYTHONPATH=. py.test InitCIMInstanceName -s            # Run only this test class
-    $ PYTHONPATH=. py.test -k InitCIMInstanceName or Bla -s  # py.test -k expressions are possible
+    testsuite
+     +-- unittest            Unit tests for single modules of the package
+     |    +-- pywbem              Unit tests for the pywbem package
+     |    +-- pywbem_mock         Unit tests for the pywbem_mock package
+     |    +-- unittest_utils      Unit tests for testsuite/unittest/utils
+     |    +-- functiontest        Unit tests for testsuite/functiontest
+     |    +-- end2endtest_utils   Unit tests for testsuite/end2endtest/utils
+     |    +-- servers             Unit tests for testsuite/servers
+     |    +-- utils               Utility functions for use by unit tests
+     +-- functiontest        Function tests for the entire pywbem client
+     +-- end2endtest         End2end tests against WBEM servers
+     |    +-- utils               Utility functions for use by end2end tests
+     +-- manualtest          Manual tests
+     +-- servers             WBEM server definition file used by some tests and module for accessing it
+     +-- profiles            Simple definitions of management profiles used by some tests
```

Comment:
profiles is such a generic word. We want it to mean something like registered_profiles but that phrase  poor also. If you think of something better than profiles, fine, otherwise ignore this comment.

Reply:
Whenever we talk about it we say either "management profiles" or "registered profiles". IMO, that is explicit enough so that we can leave the directory name at `profiles` and the file name at `profiles.yml`.